### PR TITLE
LPS-74301 Limit the number of calendarIds added to each ajax request URL to avoid exceeding the URI size limit

### PR DESCRIPTION
**Pull request by dustinryerson**
Hey @natocesarrego,

I hope you're doing well!

Here's the link for this ticket: https://issues.liferay.com/browse/LPS-74301

Please let me know if you need any further information.

Thanks!

### CODE REVIEW

**Comment by natocesarrego** [comment]
Hey @dustinryerson,

I'm doing great! Thanks for asking.

Have you sent this pull to me just to get a second opinion before forward your fix the the Product Team?

Best.

**Comment by natocesarrego** [comment]
ci:stop

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by dustinryerson** [comment]
@natocesarrego yes I wanted to pass this fix through the SME first before sending to the product team, if that workflow is ok with you?

**Comment by natocesarrego** [comment]
Of course, my friend! I just would like to double check with you. 😉

Btw, from my side your pull is fine, but I'll talk to Adam this afternoon to hear his opinion.

Best.

**Comment by dustinryerson** [comment]
Sounds great! I'll await the results of your discussion with Adam.

File: modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js [hunk 1]
```diff
@@ -360,30 +360,71 @@ AUI.add(
 				_invokeResourceURL: function(params) {
 					var instance = this;
```
Comment (natocesarrego) [inline modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js]: Hey @dustinryerson,

I've just discussed with Adam about your pull and he would like to know if you've considered to fix this issue using a POST request, once a POST would avoid the URI size limitation.

File: modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js [hunk 2]
```diff
@@ -360,30 +360,71 @@ AUI.add(
...
+									on: {
+										success: function() {
+											params.callback(this.get('responseData'));
+										}
+									}
+								}
+							);
+
+							beginIndex = endIndex;
+
```
Comment (natocesarrego) [inline modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js]: @dustinryerson, another comment is regarding the use of the ternary operator. Do you think BChan will allow the use of it? Me and Adam are a bit unsure about his opinion.

**Comment by dustinryerson** [comment]
@natocesarrego thanks for your feedback. I'm hoping to have time to work on this in the next day or two.

**Comment by natocesarrego** [comment]
Ok, @dustinryerson. Take your time.

Best.

**Comment by dustinryerson** [comment]
See https://github.com/natocesarrego/liferay-portal/pull/93

**Pull request closed without merging**